app/src/main/kotlin/com/adriangl/pokeapi_mvvm/pokemonlist/PokemonListViewModel.kt

```diff
@@ -56,7 +68,10 @@ data class PokemonListItem(val name: String, val number: Int, val sprite: String
     }
 }
 
-data class PokemonListViewData(val pokemonListRes: Resource<List<PokemonListItem>>) {
+data class PokemonListViewData(
+    val pokemonListRes: Resource<List<PokemonListItem>>,
+    val filter: (PokemonListItem) -> Boolean = { true }
```

Comment:
I think the filter (at least the filter term) should be held by the Store so we don't fragment the state of the app in multiple places, as I mentioned above.

Reply:
I also think that the view should just receive the filtered list and not do anything else, so it would be ideal to apply the filter when reducing the state into the ViewData